dev/io.openliberty.microprofile.telemetry.1.0.internal_fat/fat/src/io/openliberty/microprofile/telemetry/internal_fat/apps/context/MultiThreadedContextServlet.java

```diff
@@ -0,0 +1,110 @@
+/*******************************************************************************
+ * Copyright (c) 2023 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package io.openliberty.microprofile.telemetry.internal_fat.apps.context;
+
+import static org.junit.Assert.assertEquals;
+
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import org.junit.Test;
+
+import componenttest.app.FATServlet;
+import io.opentelemetry.api.baggage.Baggage;
+import io.opentelemetry.api.baggage.BaggageBuilder;
+import io.opentelemetry.context.Context;
+import jakarta.annotation.Resource;
+import jakarta.enterprise.concurrent.ManagedExecutorService;
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.servlet.annotation.WebServlet;
+
+@SuppressWarnings("serial")
+@WebServlet("/context")
+@ApplicationScoped
+public class MultiThreadedContextServlet extends FATServlet {
+
+    @Resource
+    ManagedExecutorService managedExecutorService;
+
+    @Test
+    public void testContextTaskWrapping() throws InterruptedException, ExecutionException, TimeoutException {
+        //add some baggage to the current context
+        addBaggage("myKey1", "myValue1");
+
+        // execute my async sub-task
+        ExecutorService executorService = Context.taskWrapping(managedExecutorService);
+        Future<String> future = executorService.submit(new MyCallable());
+        String baggageValue = future.get(5, TimeUnit.SECONDS);
+        assertEquals("myValue1", baggageValue);
+
+        //check that the baggage was not modified by the sub-task
+        assertBaggage("myKey1", "myValue1");
+    }
+
+    @Test
+    public void testContextWrap() throws InterruptedException, ExecutionException, TimeoutException {
+        //add some baggage to the current context
+        addBaggage("myKey1", "myValue1");
+
+        // execute my async sub-task
+        Context context = Context.current();
+        ExecutorService executorService = context.wrap(managedExecutorService);
```

Comment:
It would probably be better here to test without making `context` the current context.